diff --git a/pandas/meson.build b/pandas/meson.build
--- a/pandas/meson.build
+++ b/pandas/meson.build
@@ -1,22 +1,25 @@
 incdir_numpy = run_command(py,
   [
     '-c',
     '''
 import os
-import numpy as np
-try:
-    # Check if include directory is inside the pandas dir
-    # e.g. a venv created inside the pandas dir
-    # If so, convert it to a relative path
-    incdir = os.path.relpath(np.get_include())
-except Exception:
-    incdir = np.get_include()
-print(incdir)
+print(os.environ["NUMPY_INCLUDES"])
      '''
   ],
   check: true
 ).stdout().strip()
+incdir_android = run_command(py,
+  [
+    '-c',
+    '''
+import os
+print(os.environ["PYTHON_INCLUDE_DIR"])
+    '''
+  ],
+  check: true
+).stdout().strip()

+inc_android = include_directories(incdir_android)
 inc_np = include_directories(incdir_numpy)
 inc_pd = include_directories('_libs/include')


diff --git a/pandas/_libs/meson.build b/pandas/_libs/meson.build
--- a/pandas/_libs/meson.build
+++ b/pandas/_libs/meson.build
@@ -162,7 +162,7 @@ foreach ext_name, ext_dict : libs_sources
         ext_name,
         ext_dict.get('sources'),
         cython_args: cython_args,
-        include_directories: [inc_np, inc_pd],
+        include_directories: [inc_android, inc_np, inc_pd],
         dependencies: ext_dict.get('deps', ''),
         subdir: 'pandas/_libs',
         install: true
diff --git a/pandas/_libs/tslibs/meson.build b/pandas/_libs/tslibs/meson.build
--- a/pandas/_libs/tslibs/meson.build
+++ b/pandas/_libs/tslibs/meson.build
@@ -42,7 +42,7 @@ foreach ext_name, ext_dict : tslibs_sources
         ext_name,
         ext_dict.get('sources'),
         cython_args: cython_args,
-        include_directories: [inc_np, inc_pd],
+        include_directories: [inc_android, inc_np, inc_pd],
         dependencies: ext_dict.get('deps', ''),
         subdir: 'pandas/_libs/tslibs',
         install: true,
diff --git a/pandas/_libs/window/meson.build b/pandas/_libs/window/meson.build
--- a/pandas/_libs/window/meson.build
+++ b/pandas/_libs/window/meson.build
@@ -1,7 +1,7 @@
 py.extension_module(
     'aggregations',
     ['aggregations.pyx'],
     cython_args: ['-X always_allow_keywords=true'],
-    include_directories: [inc_np, inc_pd],
+    include_directories: [inc_android, inc_np, inc_pd],
     subdir: 'pandas/_libs/window',
     override_options : ['cython_language=cpp'],
     install: true,
@@ -11,7 +11,7 @@ py.extension_module(
     'indexers',
     ['indexers.pyx'],
     cython_args: ['-X always_allow_keywords=true'],
-    include_directories: [inc_np, inc_pd],
+    include_directories: [inc_android, inc_np, inc_pd],
     subdir: 'pandas/_libs/window',
     install: true,
 )
